### PR TITLE
Add some more typing for the public api

### PR DESCRIPTION
**Pull request by rchiodo**
Was just using Pylance to test debugpy and started added some typing to the public API

### CODE REVIEW

File: src/debugpy/server/cli.py [hunk 1]
```diff
@@ -41,13 +43,14 @@
 
 class Options(object):
     mode = None
-    address = None
+    address: "Optional[tuple[str, int]]" = None
```
Comment (debonte) [inline src/debugpy/server/cli.py:46]: I think the general feeling in the typing community is that `Foo | None` is more readable than `Optional[Foo]`.

https://typing.readthedocs.io/en/latest/source/modernizing.html#typing-union-and-typing-optional

Comment (rchiodo) [reply src/debugpy/server/cli.py:46]: I'll change them to that then.

File: src/debugpy/server/cli.py [hunk 2]
```diff
@@ -41,13 +43,14 @@
 
 class Options(object):
     mode = None
-    address = None
+    address: "tuple[str, int] | None" = None
```
Comment (debonte) [inline src/debugpy/server/cli.py:46]: I think the quotes are unnecessary?
```suggestion
    address: tuple[str, int] | None = None
```

**Review by debonte — APPROVED** [review]

File: src/debugpy/server/cli.py (hunk 2 above)
Comment (rchiodo) [reply src/debugpy/server/cli.py:46]: They're only unnecessary on newer Python versions. Debugpy has to run on 3.8 I believe.

image

Comment (debonte) [reply src/debugpy/server/cli.py:46]: Oh, interesting, I haven't run into that before.

Comment (rchiodo) [reply src/debugpy/server/cli.py:46]: Yeah debugpy's pyproject forces 3.8:

```
[tool.pyright]
pythonVersion = "3.8"
```

**Pull request merged**